lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java

```diff
@@ -131,7 +132,12 @@ public Tree parse(CharSequence input) {
     var metadata = getMetadata(state);
     serializedTree.order(ByteOrder.LITTLE_ENDIAN);
     var message = new Message(serializedTree, input, base, metadata);
-    return Tree.deserialize(message);
+    try {
+      return Tree.deserialize(message);
+    } catch (BufferUnderflowException | IllegalArgumentException e) {
+      System.err.println("Unrecoverable parser failure for: " + input);
```

Comment:
Can we use our logging framework, please?

Reply:
Fair enough